From 7ea1dd1da919bc5d29fdbf631408957af13966ed Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Fri, 6 Mar 2015 13:52:46 +0530
Subject: [PATCH] XPWR: Fixed pmic irq miss issue on system suspend.

PMIC interrupts are disabled on system suspend and
reenabled on resume. But SOC is missing pmic interrupts raised
during suspend. This issue is fixed by add NO_SUSPEND flag to pmic
interrupt request.

Change-Id: Ic5b4c706f7d5b983cd471fd35c4c3a7e35ddf0c5
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1157
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/extcon/extcon-dc-pwrsrc.c | 3 ++-
 drivers/mfd/intel_soc_pmic_dc.c   | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/extcon/extcon-dc-pwrsrc.c b/drivers/extcon/extcon-dc-pwrsrc.c
index 797a9a4..ca1feb4 100644
--- a/drivers/extcon/extcon-dc-pwrsrc.c
+++ b/drivers/extcon/extcon-dc-pwrsrc.c
@@ -537,7 +537,8 @@ static int dc_xpwr_pwrsrc_probe(struct platform_device *pdev)
 		info->irq[i] = platform_get_irq(pdev, i);
 		ret = request_threaded_irq(info->irq[i],
 				NULL, dc_xpwr_pwrsrc_isr,
-				IRQF_ONESHOT, PWRSRC_DRV_NAME, info);
+				IRQF_ONESHOT | IRQF_NO_SUSPEND,
+				PWRSRC_DRV_NAME, info);
 		if (ret) {
 			dev_err(&pdev->dev, "request_irq fail :%d err:%d\n",
 							info->irq[i], ret);
diff --git a/drivers/mfd/intel_soc_pmic_dc.c b/drivers/mfd/intel_soc_pmic_dc.c
index e15fe0b..5820640 100644
--- a/drivers/mfd/intel_soc_pmic_dc.c
+++ b/drivers/mfd/intel_soc_pmic_dc.c
@@ -625,7 +625,8 @@ static int dollar_cove_init(void)
 
 struct intel_soc_pmic dollar_cove_pmic = {
 	.label		= "dollar cove",
-	.irq_flags	= IRQF_TRIGGER_FALLING | IRQF_ONESHOT,
+	.irq_flags	= IRQF_TRIGGER_FALLING | IRQF_ONESHOT
+				| IRQF_NO_SUSPEND,
 	.init		= dollar_cove_init,
 	.cell_dev       = dollar_cove_dev,
 	.irq_regmap	= dollar_cove_irqregmap,
-- 
1.9.1

